### PR TITLE
Add L3 short test for RN50 training

### PR DESCRIPTION
**Pull request by JanuszL**
Signed-off-by: Janusz Lisiecki <jlisiecki@nvidia.com>

## Category:
**Other** (*e.g. Documentation, Tests, Configuration*)

## Description:
 -adds L3 short test for RN50 training

### Affected modules and functionalities:
- QA scripts

### Tests:

- [ ] Existing tests apply
- [x] New tests added
  - [x] Python tests
    - TL3_RN50_short
  - [ ] GTests
  - [ ] Benchmark
  - [ ] Other
- [ ] N/A

### Documentation
- [ ] Existing documentation applies
- [ ] Documentation updated
  - [ ] Docstring
  - [ ] Doxygen
  - [ ] RST
  - [ ] Jupyter
  - [ ] Other
- [x] N/A

#### Requirements
- [ ] Implements new requirements
- [ ] Affects existing requirements
- [x] N/A

**REQ IDs**: N/A

**JIRA TASK**: N/A

### CODE REVIEW

**Comment by stiepan** [comment]
Why do we need a short test when we have the full convergence test?

**Comment by JanuszL** [comment]
For platforms with a smaller number of GPUs - like SBSa runners (like 2xGPU).

File: qa/TL3_RN50_convergence/test_mxnet.sh [hunk 1]
```diff
@@ -10,13 +10,13 @@ python /opt/mxnet/example/image-classification/train_imagenet_runner \
...
-cat dali.log  | grep -o "Speed: [0-9]*\.[0-9]*" > tmp3.log
+cat dali.log  | grep -o "Speed: [0-9]*\.[0-9]*" | grep -o "[0-9]*\.[0-9]*" > tmp2.log
 cat tmp2.log | grep -o "0\.[0-9]*" > dali.log
-cat tmp3.log | grep -o "[0-9]*\.[0-9]*" > tmp2.log
 
 best=`awk 'BEGIN { max = -inf } { if ($1 > max) { max = $1 } } END { print max }' dali.log`
 mean=`awk 'BEGIN { sum = 0; n = 0 } { sum += $1; n += 1 } END { print sum / n }' tmp2.log`
 
+rm tmp2.log
 rm tmp2.log tmp3.log
```
Comment (mzient) [inline qa/TL3_RN50_convergence/test_mxnet.sh:20]: ?

File: qa/TL3_RN50_convergence/test_mxnet.sh [hunk 2]
```diff
@@ -10,13 +10,13 @@ python /opt/mxnet/example/image-classification/train_imagenet_runner \
...
 cat dali.log  | grep -o "Validation-accuracy=0\.[0-9]*" > tmp2.log
-cat dali.log  | grep -o "Speed: [0-9]*\.[0-9]*" > tmp3.log
+cat dali.log  | grep -o "Speed: [0-9]*\.[0-9]*" | grep -o "[0-9]*\.[0-9]*" > tmp2.log
 cat tmp2.log | grep -o "0\.[0-9]*" > dali.log
-cat tmp3.log | grep -o "[0-9]*\.[0-9]*" > tmp2.log
 
 best=`awk 'BEGIN { max = -inf } { if ($1 > max) { max = $1 } } END { print max }' dali.log`
 mean=`awk 'BEGIN { sum = 0; n = 0 } { sum += $1; n += 1 } END { print sum / n }' tmp2.log`
 
+rm tmp2.log
```
[the remark this answers is not in the document]
Comment (JanuszL) [reply qa/TL3_RN50_convergence/test_mxnet.sh]: Fixed

**Review by mzient — APPROVED** [review]

File: qa/TL3_RN50_convergence/test_mxnet.sh [hunk 3]
```diff
@@ -10,14 +10,13 @@ python /opt/mxnet/example/image-classification/train_imagenet_runner \
        -n $NUM_GPUS --seed 42 2>&1 | tee dali.log
 
 cat dali.log  | grep -o "Validation-accuracy=0\.[0-9]*" > tmp2.log
-cat dali.log  | grep -o "Speed: [0-9]*\.[0-9]*" > tmp3.log
+cat dali.log  | grep -o "Speed: [0-9]*\.[0-9]*" | grep -o "[0-9]*\.[0-9]*" > tmp2.log
```
[the remark this answers is not in the document]
Comment (JanuszL) [reply qa/TL3_RN50_convergence/test_mxnet.sh]: Fixed. Good catch.

**Review by stiepan — APPROVED** [review]

File: qa/TL3_RN50_short/test_pytorch.sh [hunk 4]
```diff
@@ -0,0 +1,60 @@
...
+   ln -sf /data/imagenet/val-jpeg/ val
+fi
+if [ ! -d "train" ]; then
+   ln -sf /data/imagenet/train-jpeg/ train
+fi
+
+LOG=dali.log
+
+SECONDS=0
+torchrun --nproc_per_node=${NUM_GPUS} main.py -a resnet50 --b 256 --loss-scale 128.0 --workers 8 --lr=0.4 --fp16-mode --epochs 5 ./ 2>&1 | tee $LOG
```
Comment (stiepan) [inline qa/TL3_RN50_short/test_pytorch.sh:25]: Is it intentionall to use twice as big batch size and number of workers compared to convergence case?

Comment (JanuszL) [reply qa/TL3_RN50_short/test_pytorch.sh:25]: The GPUs we will get for this test are 40GB+ so this batch should fit just fine.

**Review by banasraf — APPROVED** [review]

**Pull request merged**